examples/shrink_grow_layout.rs

```diff
@@ -0,0 +1,108 @@
+use bevy::{
+    prelude::{App as BevyApp, AssetServer, Commands, Res, ResMut},
+    window::WindowDescriptor,
+    DefaultPlugins,
+};
+use kayak_core::{
+    styles::{LayoutType, Style, StyleProp, Units},
+    OnLayout,
+};
+use kayak_core::{Color, EventType, OnEvent};
+use kayak_render_macros::use_state;
+use kayak_ui::core::{render, rsx, widget, Index};
+use kayak_ui::widgets::{App, Text, Window};
+use kayak_ui::{
+    bevy::{BevyContext, BevyKayakUIPlugin, FontMapping, UICameraBundle},
+    widgets::Button,
+};
+
+#[widget]
+fn GrowShrink() {
+    let (width, set_width, _) = use_state!(150.0);
+
+    let button_styles = Style {
+        width: StyleProp::Value(Units::Pixels(100.0)),
+        height: StyleProp::Value(Units::Pixels(24.0)),
+        layout_type: StyleProp::Value(LayoutType::Row),
+        background_color: StyleProp::Value(Color::new(0.33, 0.33, 0.33, 1.0)),
+        ..Default::default()
+    };
+
+    let fill = Style {
+        width: StyleProp::Value(Units::Pixels(width)),
+        height: StyleProp::Value(Units::Pixels(28.0)),
+        layout_type: StyleProp::Value(LayoutType::Column),
+        background_color: StyleProp::Value(Color::new(1.0, 0.0, 0.0, 1.0)),
+        ..Default::default()
+    };
+
+    let grow_fn = set_width.clone();
+    let shrink_fn = set_width.clone();
+
+    let grow = OnEvent::new(move |_, event| match event.event_type {
+        EventType::Click(..) => grow_fn(width + 10.0),
```

Comment:
To help drive the point home that the displayed width is coming from the layout event itself and not the `width` state variable, maybe we add the `rand` crate as a dev-dependency and use that to determine how much to grow/shrink by (instead of just `10.0`). Thoughts?